USAGOV-762-Mobile-Nav-Items-on-Special-Pages

cwacht opened the pull request:
<!--- Provide a general summary of your changes in the title above -->
## Jira Task
<!--- Provide a link to the Jira ticket -->
https://cm-jira.usa.gov/browse/USAGOV-762

## Description
<!--- Describe your changes in detail -->
Updated the Mobile nav for Wizard, Directory Record, Error, and Elected Officials pages.
On my local environment, I often had to clear my cache to get the correct nav items to appear.
Need to move error pages to the root of the menu structure.

## Testing
This change focuses on displaying the proper items in the mobile nav on these pages.

### Error Pages:
/page-error
/search-error

### Elected Officials Result Pages (start from the form on /elected-officials)
/elected-officials-results
/elected-officials-email

### Spanish Elected Officials Result Pages (start from the form on /es/funcionarios-electos)
/es/funcionarios-electos-resultados
/es/funcionarios-electos-correo-electronico

### Directory Records Listing Pages
/agency-index
/es/indice-agencias

### Directory Records
/agencies/us-abilityone-commission (and others)
es/agencias/administracion-de-alimentos-y-medicamentos (and others)

Please also test the mobile menu items on a variety of other pages to ensure that they are still working as expected.

## Checklist for the Developer
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask for help! -->
- [ ] A link to the JIRA ticket has been added above.
- [ ] The JIRA ticket identifies the desired result of this change.
- [ ] The JIRA ticket contains clear acceptance criteria.
- [ ] The JIRA ticket contains clear testing steps.
- [ ] The JIRA ticket contains a description of "Done".
- [ ] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [ ] The branch name of this PR matches the project standards.
- [ ] QA steps are followed and any changes to process are updated in the ticket.
- [ ] Code Standards are followed, there are no bad practices in use.
- [ ] Config changes (if any) include only necessary changes.
- [ ] No errors in Drupal or Client Browser.
- [ ] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [ ] There are no known side-effects outside of expected behavior.

cwacht committed:
USAGOV-762-Mobile-Nav-Items-on-Special-Pages: Wizard, Directory Record, and Elected Officials pages

cwacht committed:
USAGOV-762-Mobile-Nav-Items-on-Special-Pages: Error Pages

cwacht committed:
USAGOV-762-Mobile-Nav-Items-on-Special-Pages: While on the disabled page, show its siblings

akf committed:
Merge branch 'dev' into USAGOV-762-Mobile-Nav-Items-on-Special-Pages

akf commented on .docker/Dockerfile-minio:
```diff
@@ -1,4 +1,4 @@
-FROM minio/minio:latest
```
I was hesitant about making this change without checking in with the other devs first, but I'm leaning toward it being the right thing to do. Just not sure it belongs in this PR...

akf commented on web/themes/custom/usagov/templates/block--mobile-navigation.html.twig:
```diff
@@ -56,6 +64,41 @@
   {% endfor %}
 {% endfor %}
 
+{% if node.type[0].value.target_id == "directory_record" %}
```
Nice! I like this better than having the HTML hard-coded the way we did for the left nav.

akf replied:
Although ... that might be why we don't have the caching problem on main nav.

akf reviewed (changes requested):
I'm also having to refresh my cache. We will need to get that figured out before we can merge this in. Based on what we did to fix the caching issue for Spanish, I think this happens when the block layout doesn't give Drupal enough clues about what to cache. This case is subtler, though -- you're overriding the menus (fine!) in a few places and that's where the caching problem happens. 

Beyond that, the menus for Federal agencies look the way I expect, and the menus on error pages look good. I think the Elected Officials menus are incorrect, as I noted in Jira.

akf committed:
USAGOV-762-Mobile-Nav-caching-fixes: Make separate blocks for mobile menus with different caching rules.

The mobile menu is usually a rendering of either the "Left Menu
English" or "Left Menu Spanish" menu. However, for (federal) Directory Records,
the "menu" is a hard-coded structure PLUS the title of the current page. So for
Directory Records, the mobile menu must appear in a block that is based on the
current page and language, not on the left nav menu and language. This enables
Drupal to cache the block appropriately.

cwacht committed:
USAGOV-762-Mobile-Nav-Items-on-Special-Pages: undo minio version change

cwacht committed:
Merge branch 'USAGOV-762-Mobile-Nav-Items-on-Special-Pages' into USAGOV-762-Mobile-Nav-caching-fixes

cwacht committed:
Merge pull request #724 from usagov/USAGOV-762-Mobile-Nav-caching-fixes

USAGOV-762: Make separate blocks for mobile menus with different caching rules.

cwacht committed:
Merge branch 'dev' into USAGOV-762-Mobile-Nav-Items-on-Special-Pages

akf committed:
Merge branch 'dev' into USAGOV-762-Mobile-Nav-Items-on-Special-Pages

akf reviewed (approved):
Looks great!
